Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp

```diff
@@ -393,6 +424,222 @@ static void webkitPrintOperationPrintPagesForFrame(WebKitPrintOperation* printOp
 #endif
 }
 
+static GRefPtr<GtkPrinter> findFilePrinter(GtkPrintSettings* settings)
+{
+    struct FindPrinterData {
+        GRefPtr<GtkPrinter> printer;
+    } data = { nullptr };
+
+    gtk_enumerate_printers([](GtkPrinter* printer, gpointer userData) -> gboolean {
+        auto& data = *static_cast<FindPrinterData*>(userData);
+        auto* backend = gtk_printer_get_backend(printer);
+        if (!g_strcmp0(G_OBJECT_TYPE_NAME(backend), "GtkPrintBackendFile")) {
+            data.printer = printer;
+            return TRUE;
+        }
+        return FALSE;
+    }, &data, nullptr, TRUE);
+
+    return data.printer;
+}
+
+static void webkitPrintOperationSendPagesToPrintPortal(WebKitPrintOperation* printOperation, WebFrameProxy* webFrame, GtkPrintSettings* printSettings, GtkPageSetup* pageSetup)
+{
+    auto* priv = printOperation->priv;
+
+    RELEASE_ASSERT(priv->portalProxy);
+    RELEASE_ASSERT(priv->preparePrintResponse && priv->preparePrintResponse->token());
+    RELEASE_ASSERT(priv->preparePrintResponse->result() == WEBKIT_PRINT_OPERATION_RESPONSE_PRINT);
+
+    const char* applicationName = g_get_application_name();
+    GUniquePtr<char> jobName(g_strdup_printf("%s job #%u", applicationName ? applicationName : "WebKit", ++jobNumber));
+    GRefPtr<GtkPrinter> filePrinter = findFilePrinter(printSettings);
+
+    // Print the page into a temporary file
```

Comment:
```suggestion
    // Print the page into a temporary file.
```

Yeah, we have rules about using complete sentences in comments. I'm not sure one extra character really adds value, but you'll get used to it....